include/tvm/task_dynamics/Clamped.h

```diff
@@ -0,0 +1,190 @@
+    Bounds max_;
+  };
+
+  template<class TD>
+  inline Clamped<TD>::Clamped(const TD& innerTaskDynamics, double max)
+    : Clamped<TD>(innerTaskDynamics, -max, max)
+  {
+  }
+
+  template<class TD>
+  inline Clamped<TD>::Clamped(const TD& innerTaskDynamics, double min, double max)
+    : innerTaskDynamics_(innerTaskDynamics)
+    , min_(min)
+    , max_(max)
+  {
+    if (min > 0)
+    {
+      throw std::runtime_error("[task_dynamics::Clamped] Minimum values must be negative.");
+    }
+    if (max < 0)
+    {
+      throw std::runtime_error("[task_dynamics::Clamped] Maximum values must be positive.");
+    }
+  }
+  
+  template<class TD>
+  inline Clamped<TD>::Clamped(const TD& innerTaskDynamics, const VectorConstRef& max)
+    : Clamped<TD>(innerTaskDynamics, -max, max)
+  {
+  }
+
+  template<class TD>
+  inline Clamped<TD>::Clamped(const TD& innerTaskDynamics, const VectorConstRef& min, const VectorConstRef& max)
+    : innerTaskDynamics_(innerTaskDynamics)
+    , min_(min)
+    , max_(max)
+  {
+    if (min.size() !=  max.size())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped] The minimum and maximum must have the same size.");
+    }
+    if ((min.array() > 0).any())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped] Minimum values must be negative.");
+    }
+    if ((max.array() < 0).any())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped] Maximum values must be positive.");
+    }
+  }
+  
+  template<class TD>
+  inline std::unique_ptr<abstract::TaskDynamicsImpl> Clamped<TD>::impl_(FunctionPtr f, constraint::Type t, const Eigen::VectorXd& rhs) const
+  {
+    return std::unique_ptr<abstract::TaskDynamicsImpl>(new Impl(f, t, rhs, innerTaskDynamics_, min_, max_));
+  }
+
+  template<class TD>
+  inline Order Clamped<TD>::order_() const
+  {
+    return innerTaskDynamics_.order();
+  }
+
+  template<class TD>
+  inline Clamped<TD>::Impl::Impl(FunctionPtr f, constraint::Type t, const Eigen::VectorXd& rhs, const TD& innerTaskDynamics, const Bounds& min, const Bounds& max)
+    : TaskDynamicsImpl(innerTaskDynamics.order(), f, t, rhs)
+    , innerTaskDynamicsImpl_(static_cast<typename TD::Impl*>(innerTaskDynamics.impl(f, t, rhs).release()))
+    , min_(min.index() == 1 ? mpark::get<Eigen::VectorXd>(min) : VectorXd::Constant(f->size(), mpark::get<double>(min)))
+    , max_(max.index() == 1 ? mpark::get<Eigen::VectorXd>(max) : VectorXd::Constant(f->size(), mpark::get<double>(max)))
+  {
+    if (min_.size() != f->size() || max_.size() != f->size())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped::Impl] Sizes of the minimum, maximum and function must be the same.");
+    }
+    if ((min_.array() > 0).any())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped::Impl] Minimum values must be negative.");
+    }
+    if ((max_.array() < 0).any())
+    {
+      throw std::runtime_error("[task_dynamics::Clamped::Impl] Maximum values must be positive.");
+    }
```

Comment:
The general problem is that it's possible to call `MyTaskDynamics::Impl::Impl` directly (even if that's not the usage for now) so that if there was a single place to perform this kind of tests, it would be there. But at the same time it's nice to check in the ctor of `MyTaskDynamics` to get the potential error at the place it would usually been made.

A workaround would be to make the ctor of `Impl` private and have a friendship declaration with `MyTaskDynamics`, but I'm reluctant to that because these two classes are already quite verbose as is. The tests themselves are not very costly.